core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/impl/QueryJoinOptimizer.java

```diff
@@ -393,6 +393,12 @@ protected TupleExpr selectNextTupleExpr(List<TupleExpr> expressions, Map<TupleEx
 			return result;
 		}
 
+		@Deprecated
+		protected double getTupleExprCardinality(TupleExpr tupleExpr, Map<TupleExpr, Double> cardinalityMap,
```

Comment:
This is one of the many reasons why I want stop using `protected` as a modifier, as much as possible.